other/docker/windows/build_toxcore.sh

```diff
@@ -4,165 +4,184 @@ set -e -x
-        export WINEPATH=`cd /usr/lib/gcc/${WINDOWS_TOOLCHAIN}/*posix/ ; winepath -w $(pwd)`\;`winepath -w /usr/${WINDOWS_TOOLCHAIN}/lib/`\;`winepath -w ${EXTRA_DEP_PREFIX_DIR}/bin`
-
-        if [ "${ALLOW_TEST_FAILURE}" = "true" ]; then
-            set +e
-        fi
-        cmake --build . --target test -- ARGS="-j50"
-        if [ "${ALLOW_TEST_FAILURE}" = "true" ]; then
-            set -e
-        fi
-    fi
+    winecfg
+    export CTEST_OUTPUT_ON_FAILURE=1
+    # add libgcc_s_sjlj-1.dll libwinpthread-1.dll into PATH env var of wine
+    export WINEPATH="$(
+      cd /usr/lib/gcc/"$WINDOWS_TOOLCHAIN"/*posix/
+      winepath -w "$PWD"
+    )"\;"$(winepath -w /usr/"$WINDOWS_TOOLCHAIN"/lib/)"
 
-    # move static dependencies
-    cp "${STATIC_TOXCORE_PREFIX_DIR}"/* "${RESULT_PREFIX_DIR}" -R
-    cp "${DEP_PREFIX_DIR}"/* "${RESULT_PREFIX_DIR}" -R
-
-    # make libtox.dll
-    cd "${SHARED_TOXCORE_PREFIX_DIR}"
-    for archive in ${STATIC_TOXCORE_PREFIX_DIR}/lib/libtox*.a
-    do
-        ${WINDOWS_TOOLCHAIN}-ar xv ${archive}
-    done
-    ${WINDOWS_TOOLCHAIN}-gcc -Wl,--export-all-symbols \
-                             -Wl,--out-implib=libtox.dll.a \
-                             -shared \
-                             -o libtox.dll \
-                             *.obj \
-                             ${STATIC_TOXCORE_PREFIX_DIR}/lib/*.a \
-                             ${DEP_PREFIX_DIR}/lib/*.a \
-                             /usr/${WINDOWS_TOOLCHAIN}/lib/libwinpthread.a \
-                             -liphlpapi \
-                             -lws2_32 \
-                             -static-libgcc
-    cp libtox.dll.a ${RESULT_PREFIX_DIR}/lib
-    mkdir -p ${RESULT_PREFIX_DIR}/bin
-    cp libtox.dll ${RESULT_PREFIX_DIR}/bin
-
-    rm -rf /tmp/*
-
-    # remove everything from include directory except tox headers
-    mv ${RESULT_PREFIX_DIR}/include/tox ${RESULT_PREFIX_DIR}/tox
-    rm -rf ${RESULT_PREFIX_DIR}/include/*
-    mv ${RESULT_PREFIX_DIR}/tox ${RESULT_PREFIX_DIR}/include/tox
-
-    sed -i "s|^prefix=.*|prefix=${RESULT_PREFIX_DIR}|g" ${RESULT_PREFIX_DIR}/lib/pkgconfig/*.pc
-    sed -i "s|^libdir=.*|libdir=${RESULT_PREFIX_DIR}/lib|g" ${RESULT_PREFIX_DIR}/lib/*.la
+    if [ "$ALLOW_TEST_FAILURE" = "true" ]; then
+      set +e
+    fi
+    cmake --build . --target test -- ARGS="-j50"
+    if [ "$ALLOW_TEST_FAILURE" = "true" ]; then
+      set -e
+    fi
+  fi
+
+  # move static dependencies
+  cp -a "$STATIC_TOXCORE_PREFIX_DIR"/* "$RESULT_PREFIX_DIR"
+  cp -a "$DEP_PREFIX_DIR"/* "$RESULT_PREFIX_DIR"
+
+  # make libtox.dll
+  cd "$SHARED_TOXCORE_PREFIX_DIR"
+  for archive in "$STATIC_TOXCORE_PREFIX_DIR"/lib/libtox*.a; do
+    "$WINDOWS_TOOLCHAIN"-ar xv "$archive"
+  done
+
+  if [ "$CROSS_COMPILE" = "true" ]; then
+    LIBWINPTHREAD="/usr/$WINDOWS_TOOLCHAIN/lib/libwinpthread.a"
+  else
+    LIBWINPTHREAD="/usr/$WINDOWS_TOOLCHAIN/sys-root/mingw/lib/libwinpthread.a"
+  fi
```

Comment:
Ah, now I see why you were complaining about the removal of `$CROSS_COMPILE` things. You have made a mistake in your initial commit which I didn't notice, which then caused this confusion about the removal.

In your initial commit [this if-statement's `then` and `else` bodies are swapped](https://github.com/nurupo/InsertProjectNameHere/commit/8a1f149bd0cf94686842b522bf8703e5b435474f#diff-86b73a62db12c39395a4ef2111140e6b50743f43e8202f3d745a204f9b9042eeR116-R120), i.e.:

```C
  if [ "${CROSS_COMPILE}" = "true" ]; then
    LIBWINPTHREAD="/usr/${WINDOWS_TOOLCHAIN}/sys-root/mingw/lib/libwinpthread.a"
  else
    LIBWINPTHREAD="/usr/${WINDOWS_TOOLCHAIN}/lib/libwinpthread.a"
  fi
```

which didn't work for me when testing cross-compiling a few days ago, as the correct path when cross-compiling is `/usr/${WINDOWS_TOOLCHAIN}/lib/libwinpthread.a` - the same path you have specified for the Cygwin build. So since they are the same path there was no point in that if-statement and I have removed it. Now I see that you meant to have those two bodies swapped.

Reply:
Yeah, I'm sorry, noticed that too.